### PR TITLE
Add proselint for mail filetype

### PR DESCRIPTION
**Pull request by KyleOndy**
Very similar to the [text](https://github.com/neomake/neomake/blob/master/autoload/neomake/makers/ft/text.vim) filetype maker, but for the mail filetype

### CODE REVIEW

File: autoload/neomake/makers/ft/mail.vim [hunk 1]
```diff
@@ -0,0 +1,9 @@
+function! neomake#makers#ft#mail#EnabledMakers()
+    return ['proselint']
+endfunction
+
+function! neomake#makers#ft#mail#proselint()
+    return {
+                \ 'errorformat': '%f:%l:%c: %m'
```
Comment (blueyed) [inline autoload/neomake/makers/ft/mail.vim]: Should specify a type explicitly, e.g. by prefixing it with `%E` for errors.

Also please add " abort" to the function definitions (check ":Neomake vint" ;)).

**Comment by KyleOndy** [comment]
A lot of the current makers don't have explicit types or abort on their functions.

Should they? If so I can put in some PRs.

**Comment by blueyed** [comment]
> Should they? If so I can put in some PRs.

Yes, the linter complains.
It gets currently done only with changed codes.
If you create a PR, that would cause conflicts for existing / pending ones probably.

**Comment by blueyed** [comment]
> I believe a warning makes more sense then a hard error. Prose is subjective

:+1: 
See also https://github.com/neomake/neomake/issues/584.

**Comment by blueyed** [comment]
Thanks!

**Comment by blueyed** [comment]
Please add it to the wiki: https://github.com/neomake/neomake/wiki/Makers

**Comment by KyleOndy** [comment]
Added to wiki. Thanks for the fast and easy PR.

**Pull request merged**